### Step 1. 필요한 라이브러리 임포트

In [1]:
import pandas as pd

### Step 2. [데이터](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user) 확인

- user_id 유저 아이디
- age 나이
- gender 성별
- occupation 직업
- zip_code 우편번호

### Step 3. 

**read_csv를 통해 데이터를 올바른 형태로 불러온 뒤 users에 저장해주세요.**  
- data : occupation.tsv
- sep ='|'
- user_id를 인덱스로 설정해주세요.

In [22]:
occupation = pd.read_csv('C:/STUDY/StudyPandas/data/occupation.tsv',sep='|')
occupation2=occupation.set_index('user_id',append=False, drop=True)
occupation2

age gender     occupation zip_code
user_id                                    
1         24      M     technician    85711
2         53      F          other    94043
3         23      M         writer    32067
4         24      M     technician    43537
5         33      F          other    15213
...      ...    ...            ...      ...
939       26      F        student    33319
940       32      M  administrator    02215
941       20      M        student    97229
942       48      F      librarian    78209
943       22      M        student    77841

[943 rows x 4 columns]

### Step 4. 초반 25행 조회

In [23]:
occupation2[:30]

age gender     occupation zip_code
user_id                                    
1         24      M     technician    85711
2         53      F          other    94043
3         23      M         writer    32067
4         24      M     technician    43537
5         33      F          other    15213
6         42      M      executive    98101
7         57      M  administrator    91344
8         36      M  administrator    05201
9         29      M        student    01002
10        53      M         lawyer    90703
11        39      F          other    30329
12        28      F          other    06405
13        47      M       educator    29206
14        45      M      scientist    55106
15        49      F       educator    97301
16        21      M  entertainment    10309
17        30      M     programmer    06355
18        35      F          other    37212
19        40      M      librarian    02138
20        42      F      homemaker    95660
21        26      M         writer    30068
22        25      M         writer    40206
23        30      F         artist    48197
24        21      F         artist    94533
25        39      M       engineer    55107
26        49      M       engineer    21044
27        40      F      librarian    30030
28        32      M         writer    55369
29        41      M     programmer    94043
30         7      M        student    55436

### Step 5. 마지막 10행 조회

In [27]:
occupation2[-10:]

age gender     occupation zip_code
user_id                                    
934       61      M       engineer    22902
935       42      M         doctor    66221
936       24      M          other    32789
937       48      M       educator    98072
938       38      F     technician    55038
939       26      F        student    33319
940       32      M  administrator    02215
941       20      M        student    97229
942       48      F      librarian    78209
943       22      M        student    77841

### Step 6. 데이터셋은 몇 개의 행으로 이루어져 있나요?

943

In [28]:
len(occupation2)

943

### Step 7.  데이터셋은 몇 개의 열로 이루어져 있나요?

4

In [39]:
len(occupation2.columns)

4

### Step 8. 열(column)의 이름들을 모두 조회해주세요

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

In [29]:
occupation2.columns

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

### Step 9. 데이터셋의 인덱스를 조회해주세요

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

In [32]:
occupation2.index

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

### Step 10. 각 열의 데이터타입은 무엇인가요?

In [34]:
occupation2.dtypes

age            int64
gender        object
occupation    object
zip_code      object
dtype: object

### Step 11. occupation column만 조회해주세요

In [44]:
occupation2['occupation']

user_id
1         technician
2              other
3             writer
4         technician
5              other
           ...      
939          student
940    administrator
941          student
942        librarian
943          student
Name: occupation, Length: 943, dtype: object

### Step 12. 데이터셋에는 몇 종류의 직업이 존재하나요?

21

In [46]:
len(occupation2['occupation'].unique())

21

### Step 13. 데이터셋에서 가장 많이 존재하는 직업은 무엇인가요?

student

In [88]:
occupation_value = pd.DataFrame(occupation2['occupation'].value_counts())
max_value = (occupation_value.occupation == occupation_value.occupation.max())
occupation_value[max_value]


occupation
student         196

### Step 14. users.describe() 코드를 실행해서 숫자 열의 요약정보를 조회해주세요

In [48]:
occupation2['age'].describe()

count    943.000000
mean      34.051962
std       12.192740
min        7.000000
25%       25.000000
50%       31.000000
75%       43.000000
max       73.000000
Name: age, dtype: float64

### Step 15. user들의 평균 나이는 몇 살인가요?
34

In [43]:
occupation2['age'].mean()

34.05196182396607

### Step 16. 데이터셋에서 각각 여성과 남성 직업 순위 Top 5를 표시해주세요

* F  
occupation  
student          60  
administrator    36  
other            36  
librarian        29  
educator         26  
* M  
occupation  
student       136  
educator       69  
other          69  
engineer       65  
programmer     60  

In [103]:
Female = occupation2.gender=='F'
F_occupation = occupation2[Female]
F_occupation

age gender  occupation zip_code
user_id                                 
2         53      F       other    94043
5         33      F       other    15213
11        39      F       other    30329
12        28      F       other    06405
15        49      F    educator    97301
...      ...    ...         ...      ...
925       18      F    salesman    49036
930       28      F   scientist    07310
938       38      F  technician    55038
939       26      F     student    33319
942       48      F   librarian    78209

[273 rows x 4 columns]

In [105]:
Male = occupation2.gender=='M'
M_occupation =occupation2[Male]
M_occupation

age gender     occupation zip_code
user_id                                    
1         24      M     technician    85711
3         23      M         writer    32067
4         24      M     technician    43537
6         42      M      executive    98101
7         57      M  administrator    91344
...      ...    ...            ...      ...
936       24      M          other    32789
937       48      M       educator    98072
940       32      M  administrator    02215
941       20      M        student    97229
943       22      M        student    77841

[670 rows x 4 columns]

In [107]:
F_rank = pd.DataFrame(F_occupation['occupation'].value_counts())
F_rank

occupation
student                60
other                  36
administrator          36
librarian              29
educator               26
writer                 19
artist                 13
healthcare             11
marketing              10
homemaker               6
programmer              6
none                    4
executive               3
scientist               3
salesman                3
engineer                2
lawyer                  2
entertainment           2
retired                 1
technician              1

In [108]:
M_rank = pd.DataFrame(M_occupation['occupation'].value_counts())
M_rank

occupation
student               136
other                  69
educator               69
engineer               65
programmer             60
administrator          43
executive              29
scientist              28
technician             26
writer                 26
librarian              22
entertainment          16
marketing              16
artist                 15
retired                13
lawyer                 10
salesman                9
doctor                  7
none                    5
healthcare              5
homemaker               1